packages/docutils/lib/logger.ts

```diff
@@ -0,0 +1,35 @@
+import consola, {Consola, ConsolaOptions} from 'consola';
+
+const rootLogger = consola.withTag('appium:docutils');
+
+const loggers = new Map<string | undefined, Consola>([[undefined, rootLogger]]);
+
+export default new Proxy(rootLogger, {
+  get(target, prop, receiver) {
+    if (prop === 'create') {
+      const create = Reflect.get(target, prop, receiver) as Consola['create'];
+      return (opts: ConsolaOptions) => {
+        const tag = opts?.defaults?.tag;
+        if (tag) {
+          if (loggers.has(tag)) {
+            return loggers.get(tag)!;
+          }
+          const child = create.call(receiver, opts);
+          loggers.set(tag, child);
+          return child;
+        }
+        return create.call(receiver, opts);
+      };
+    }
+    return Reflect.get(target, prop);
+  },
+  set(target, prop, value, receiver) {
```

Comment:
note to self: `set` is supposed to return a `boolean` depending on whether or not it succeeds.  since we are calling this handler for multiple objects at once, we only return `true` if _all_ objects return `true`.  does that make sense?  no. it's a proxy. they don't make sense by design

Reply:
this is no longer valid